text/0068-feat-login-into-portable-account-with-metamask.md

```diff
@@ -0,0 +1,116 @@
+- FIP: 68
+- title: Login into Portable account with metamask 
+- author: Sabyasachi Patra, @asabya
+- status: draft
+- created: 2023-03-01
+
+# Summary
+Addressing web3 login workflow to portable account where users can retrieve their wallet in a decentralized manner from browser with metamask.
+
+# Context, motivation and guide level explanation
+
+The solution worked out here allows to log in to the user account from any browser into the same account described in [FIP-59](https://github.com/fairDataSociety/FIPs/blob/master/text/0059-portable-account.md) created with
+[fdp-create-account](https://github.com/fairDataSociety/fdp-create-account) project.
+
+# Reference-level explanation
+
+For the login to work with metamask, user should already have a portable account created with [fdp-create-account](https://github.com/fairDataSociety/fdp-create-account).
+
+As we know while creating a portable account users can provide a 12 word mnemonic phrase for the portable wallet or a 12 word mnemonic will be generated for them.
+
+This solution starts with importing the portable wallet into metamask. After that it is a two-step process
+
+## Step 1: 
+```go
+ConnectPortableAccountWithWallet(userName, passPhrase, addressHex, signature string) error
+```
+
+This function will get the portable wallet (wallet seed) and re-upload an alternate socTopic described later
+
+## Step 2: 
+```go
+LoginWithWallet(addressHex, signature string) (*user.Info, error) 
+```
+
+This function will get the encrypted portable wallet (wallet seed) and username from the alternate socTopic and decrypt it with the signature provided by the user.
+
+
+## Uploading portable wallet (wallet seed) to Ethereum Swarm to an alternate socTopic
+
+In FIP-59 we have discussed, about the socTopic which is a topic in the swarm where the portable account is stored. In this solution we will be using an alternate socTopic, but with the same logic, to store the wallet seed.
+```
+socTopic = H(fdpLoginVersion + portableWalletAddress + signature)
+```
+
+Where `portableWalletAddress` and the `signature` will be provided by the user. As we have already imported the portable wallet into metamask, we can get the `portableWalletAddress` from there. The `signature` will be generated by signing the `portableWalletAddress` with metamask with the portable wallet.

```

Comment:
so I guess this SOC will be written by any other wallet, right? if this topic have to be calculated with the portable wallet signature then the portable wallet - that you want to retrieve - should be in the wallet handler all-the-time to produce this signature. It has a bit of contradiction.

Reply:
if one forgets the password they should use the 12 words to recreate the wallet the same way as they can import it to Metamask.